### DISCUSSION TITLE
[Feat]: Plug in problems

### DISCUSSION THREAD

**Discussion by chenhuanguang** [coder/code-server#5667] [Q&A]
I install and start the code server in the form of a package. I want to build in some vscode plug-ins. When I start the code server, these plug-ins are already in it. Because my environment has no network, so I cannot download plug-ins online. How should I download plug-ins and place them in a specified location

**Comment by jsjoeio** [comment]
You can download extensions from the Open VSX marketplace as vsix files and then install them from there. That should work!

**Reply by chenhuanguang** [reply]
Is there a better solution? I can prepare the plug-in in advance. Every time I start the code server, the plug-in will be automatically loaded, because I see that you also have some built-in plug-ins. Can you do the same

**Reply by jsjoeio** [reply]
Hmm...I can't think of something at the moment. You can change the directory where extensions are installed using `--extensions-dir` when you start code-server. So you could do that in an online environment then use that directory in your offline environment.
